api/src/main/java/org/openmrs/api/ObsService.java

```diff
@@ -536,5 +537,32 @@ public Integer getObservationCount(List<Person> whom, List<Encounter> encounters
 			List<Concept> answers, List<PERSON_TYPE> personTypes, List<Location> locations, List<Visit> visits,
 			Integer obsGroupId, Date fromDate, Date toDate, boolean includeVoidedObs, String accessionNumber)
 			throws APIException;
-	
+
+	/**
+	 * This method gets ObsReferenceRange by id
+	 * 
+	 * @param id ObsReferenceRange id
+	 * @return {@link ObsReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get the ObsReferenceRange with the given id
+	 * <strong>Should</strong> return null if ObsReferenceRange is not found
+	 */
+	@Authorized(PrivilegeConstants.GET_OBS)
+	ObsReferenceRange getObsReferenceRangeById(Integer id);
+
+	/**
+	 * This method gets the latest Obs by conceptId. The latest obs is determined by the date created.
+	 * 
+	 * @param conceptId conceptId
+	 * @return {@link ObsReferenceRange}
+	 *
+	 * @since 2.7.0
+	 *
+	 * <strong>Should</strong> get the latest Obs with the given conceptId
+	 * <strong>Should</strong> return null if Obs is not found
+	 */
+	@Authorized(PrivilegeConstants.GET_OBS)
+	Obs getLatestObsByConceptId(String conceptId);
```

Comment:
Could there be a reason why we are not using the existing Context.getObsService().getObservations()?

Reply:
This is a dedicated method to fetch Obs by conceptId. It returns one Obs unlike the other method that returns a list.